### PR TITLE
Use new snapshot deserialization model

### PR DESCRIPTION
**Pull request by ungwudik**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala [hunk 1]
```diff
@@ -221,7 +221,7 @@ class SnapshotAPISpec extends AnyFreeSpecLike with Matchers with BeforeAndAfterA
 
           // validate the snapshot was added correctly: list snapshots in Rawls, should return 1, which we just added.
           val listResponse = listSnapshotReferences(projectName, workspaceName)
-          val resources = Rawls.parseResponseAs[ResourceList](listResponse).getResources.asScala
+          val resources = Rawls.parseResponseAs[ResourceList](SnapshotListResponse).gcpDataRepoSnapshots
```
Comment (davidangb) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala]: shouldn't this be:
```
Rawls.parseResponseAs[SnapshotListResponse](listResponse).gcpDataRepoSnapshots
```

**Review by sagehen03 — APPROVED** [review]
lgtm, sorry about missing these.  It would be great if these tests would run as part of sbt test or at least upon PR/branch publication.

**Review by davidangb — APPROVED** [review]
:+1: assuming all tests pass

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala (hunk 1 above)
Comment (ungwudik) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala]: Yes, this was a copy and paste mistake after my branch-switching shenanigans. I updated and re-ran the tests locally

**Comment by ungwudik** [comment]
I see the error below in the failed jenkins test. I'm going to assume it's flakiness and rerun

`[warn] 	Note: Some unresolved dependencies have extra attributes.  Check that these dependencies exist with the requested attributes.
[warn] 		net.virtual-void:sbt-dependency-graph:0.9.2 (sbtVersion=1.0, scalaVersion=2.12)
[warn] 		ch.epfl.scala:sbt-scalafix:0.9.21 (sbtVersion=1.0, scalaVersion=2.12)
[warn] 
[warn] 	Note: Unresolved dependencies path:
[error] sbt.librarymanagement.ResolveException: Error downloading net.virtual-void:sbt-dependency-graph;sbtVersion=1.0;scalaVersion=2.12:0.9.2
[error]   Not found
[error]   Not found
[error]   download error: Caught java.net.ConnectException: Connection timed out (Connection timed out) (Connection timed out (Connection timed out)) while downloading https://repo1.maven.org/maven2/net/virtual-void/sbt-dependency-graph_2.12_1.0/0.9.2/sbt-dependency-graph-0.9.2.pom
[error]   not found: /root/.ivy2/localnet.virtual-void/sbt-dependency-graph/scala_2.12/sbt_1.0/0.9.2/ivys/ivy.xml
[error]   not found: https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/net.virtual-void/sbt-dependency-`

**Comment by ungwudik** [comment]
> lgtm, sorry about missing these. It would be great if these tests would run as part of sbt test or at least upon PR/branch publication.

Yeah I think running in PRs will be worth the investment.

**Pull request merged**